vortex-array/src/array/sparse/compute.rs

```diff
@@ -125,3 +131,115 @@ impl ScalarAtFn for SparseArray {
         }
     }
 }
+
+impl TakeFn for SparseArray {
+    fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        let indices = compute::flatten::flatten_primitive(indices)?;
+        let adjusted_indices = match_each_integer_ptype!(indices.ptype(), |$P| {
+             indices.typed_data::<$P>()
+                .iter()
+                .map(|i| *i as usize + self.indices_offset)
+                .collect::<Vec<_>>()
+        });
+
+        // TODO(robert): Use binary search instead of search_sorted + take and index validation to avoid extra work
+        let physical_indices = PrimitiveArray::from(
+            adjusted_indices
+                .iter()
+                .map(|i| {
+                    search_sorted(self.indices(), *i, SearchSortedSide::Left).map(|s| s as u64)
```

Comment:
fwiw, I found that this commit https://github.com/fulcrum-so/vortex/pull/192/commits/09fc199e29f5fb6e38743f7e9468c93f4e089534 made an enormous improvement (8x faster) for taking 10K elements from a patched bitpacked array:

```
take_10K_patches        time:   [165.73 µs 174.12 µs 184.37 µs]
                        change: [-87.543% -87.192% -86.796%] (p = 0.00 < 0.05)
                        Performance has improved.
```

Reply:
basically, for non-trivial number of indices, building the hash map is way faster  than repeatedly doing binary search

Reply:
I incorporated your method into my pr